course_is_digital.py

```diff
@@ -0,0 +1,142 @@
+    select_tag_exam_year: Optional[bs4.element.Tag] = soup.find(attrs={"id": "selectedYear"})
+    if select_tag_exam_year is None:
+        warn(f"Couldn't find the exam year select element, fagkode={course_code}, returning empty list")
+        return []
+    
+    exam_years: List[str] = [option["value"] for option in select_tag_exam_year.find_all("option")]
+    
+    return exam_years
+
+
+def retrieve_exam_type_of_years(
+        course_code: str,
+        years: List[str],
+        sleep_time_mean_ms=0.1) -> Dict[str, Dict[str, bool]]:
+    """
+    Return a dictionary that maps course codes to a dictionary of
+    "Fall" or "Spring" that maps to True or False, True if that exam
+    was digital, False otherwise
+    
+    
+    :param course_code: NTNU course code
+    :param years: list of (university) years to retrieve
+    :param sleep_time_mean_ms: average time to sleep between each https call
+    :return:
+    """
+    with requests.Session() as session:
+        session = requests.Session()
+        result = dict()
+        
+        for year in years:
+            url = URL_COURSE_YEAR_FORMAT_STRING.format(course_code, year)
+            response = session.get(url)
+            soup = bs4.BeautifulSoup(response.text, HTML_PARSER)
+            omEksamen: Optional[bs4.element.Tag] = soup.find(attrs={"id": "omEksamen"})
+            if omEksamen is None:
+                warn(f"Couldn't find omEksamen for {course_code} year {year}")
+                continue
+            dl: Optional[bs4.element.Tag] = omEksamen.find("dl")
+            if dl is None:
+                warn(f"omEksamen tag for {course_code} year {year} had no dl tag, skipping")
+                continue
+            term_is_digital_dict: Dict[str, bool] = dict()
+            for dt in dl.find_all("dt"):
+                term: Optional[bs4.element.Tag] = dt.find(class_="exam-term")
+                if term is None:
+                    continue
+                
+                system: Optional[bs4.element.Tag] = dt.find(class_="exam-system")
+                exam_status: Optional[bs4.element.Tag] = dt.find(class_="exam-code")
+                
+                # skip continuation exam
+                #is_ordinary = exam_status.find("abbr").text.strip() == "ORD"
+                #if not is_ordinary:
+                #    continue
+                
+                term_std = ""
+                term_txt = term.text.strip()
+                if term_txt == "Vår":
+                    term_std = "Spring"
+                elif term_txt == "Høst":
+                    term_std = "Fall"
+                elif term_txt == "Sommer":
+                    term_std = "Summer"
+                else:
+                    continue
+                term_is_digital_dict[term_std] = system.text.strip() == "INSPERA"
+            result[year] = term_is_digital_dict
+            # sleep a little bit to avoid hammering the website
+            time.sleep(0.05)
```

Comment:
Shoudl we use the `sleep_time_mean_ms` here?

Reply:
Probably, was testing with time.sleep